nix/lib.nix

```diff
@@ -8,25 +8,62 @@
     { config, pkgs, lib, ... }:
     let
       # Derive name from filename
-      name = lib.pipe mod [
+      service = lib.pipe mod [
         builtins.baseNameOf
         (lib.strings.splitString ".")
         builtins.head
       ];
+      serviceModule = { config, name, ... }: {
+        options = {
+          namespace = lib.mkOption {
+            description = ''
+              Namespace for the ${service} service
+            '';
+            default = "${service}.${name}";
+            type = lib.types.str;
+          };
+          outputs.defaultProcessSettings = lib.mkOption {
+            type = lib.types.deferredModule;
+            internal = true;
+            readOnly = true;
+            description = ''
+              Default settings for all processes under the ${service} service
```

Comment:
```suggestion
              Default settings for all processes under the ${service}.${name} service
```

Reply:
If `name` is dynamic, how will the docs render?

Reply:
My bad, I missed that.